NDTensors/src/zeros/zeros.jl

```diff
@@ -0,0 +1,46 @@
+struct Zeros{ElT,N,DataT} <: AbstractArray{ElT,N}
+  z::FillArrays.Zeros
+  is::Tuple
+  function NDTensors.Zeros{ElT,N,DataT}(inds::Tuple) where {ElT,N,DataT}
+    @assert eltype(DataT) == ElT
+    @assert ndims(DataT) == N
+    z = FillArrays.Zeros(ElT, dim(inds))
+    return new{ElT,N,DataT}(z, inds)
+  end
+end
```

Comment:
Okay I updated with Axes.  Right now I have it possible to make higher order `NDTensor.Zeros` types but the Alloc type does not depend on N. Also, when a `NDTensors.Zeros` is constructed in ITensors, I flatten the indices down and just store a vector (since the true indices will be stored in the `Tensor`).

Reply:
What do you mean by "the Alloc type does not depend on N"? It seems simplest to constrain it so that the `Alloc` type is some `AbstractArray` that is order `N`.